Fix SQLite version transition

JingMatrix opened the pull request:
1. Avoid adding new columns when creating database anew
2. SQLiteException has updated, whose error message now includes the error code, for example `(code 1 SQLITE_ERROR)`.

JingMatrix commented:
@USnwst Please test the CI build https://github.com/JingMatrix/LSPosed/actions/runs/14122892276

USnwst commented:
I reinstalled this LSPosed, but it still doesn't display anything. Here is the log file：

[LSPosed_2025-03-28T15_59_11.832724.zip](https://github.com/user-attachments/files/19501065/LSPosed_2025-03-28T15_59_11.832724.zip)

JingMatrix commented:
Crucial logs are
```
[ 2025-03-28T15:57:23.653        0:  2854:  2854 D/LSPosedService  ] create new datebase
[ 2025-03-28T15:57:23.654        0:  2854:  2854 E/SQLiteLog       ] (1) no such table: configs in "SELECT `group`, `key`, data FROM configs WHERE module_pkg_name = ? and user_id = ?"

```

I have no clear idea why this is happening, have you tried a complete uninstallation ?

yuche commented:
@JingMatrix 

[LSPosed_2025-03-28T22_42_35.196997.zip](https://github.com/user-attachments/files/19507705/LSPosed_2025-03-28T22_42_35.196997.zip)

I'm having the same problem(https://github.com/JingMatrix/LSPosed/issues/235), here's my log (version 7177, CI build).

I can confirm that lsposed was completely removed before installing 7177, KernelSU manager doesn't have lsposed, /data/adb doesn't have lsposed, /data/adb/modules doesn't have it either.

But my reported error seem to be different from @USnwst :

```
[ 2025-03-28T22:41:11.060     2000: 15011: 15044 E/LSPosedManager  ] Wi: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#unexpected-json-structure
	at Og.a(Unknown Source:90)
	at Br.e(Unknown Source:47)
	at B1.run(Unknown Source:9)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
	at java.util.concurrent.FutureTask.run(FutureTask.java:290)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#unexpected-json-structure
	at aj.V(Unknown Source:59)
	at aj.r(Unknown Source:28)
	at K3.b(Unknown Source:18)
	at Og.a(Unknown Source:41)
	... 7 more
```

yuche commented:
<del>Also can I install modules locally? It doesn't feel right to pull modules from a CDN, and if the supply chain is poisoned my device will be compromised.</del>

I found that by installing the apk I was able to enable the module in lsposed manager.

Dev4Mod commented:
The folder where the settings are located is /data/adb/lspd

JingMatrix commented:
@yuche Could you please explain more about your sentence `by installing the apk`? Which apk you were talking about?

Moreover, please upload your logs again. There were errors I didn't understand in your previous logs.

Stillhard commented:
> @yuche Could you please explain more about your sentence `by installing the apk`? Which apk you were talking about?

maybe he meant by installed manager apk manually

yuche commented:
My problem was solved by disabling DNS over HTTPS and I was able to pull  remote repositories. 

I've never used lsposed before, I used to think that the lsposed module was only installable from a remote CDN, but it's actually installable from anywhere you get the apk.

JingMatrix committed:
Correct database migration and initialization logic

Reworks the `initDB` function to resolve critical failures.

- Prevents a "no such table: configs" crash on fresh installs by
  deferring the compilation of CREATE TABLE statements until their
  dependencies are met.
- Fixes a "duplicate column" error by correcting the initial schema
  and letting migration logic add new columns.
- Replaces the fragile fall-through switch and nested transactions with a
  robust, sequential upgrade process inside a single atomic transaction.
